rclcpp/include/rclcpp/parameter_map.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__PARAMETER_MAP_HPP_
+#define RCLCPP__PARAMETER_MAP_HPP_
+
+#include <exception>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "rclcpp/parameter.hpp"
+#include "rclcpp/parameter_value.hpp"
+#include "rclcpp/visibility_control.hpp"
+#include "rcl_yaml_param_parser/types.h"
+
+namespace rclcpp
+{
+
+/// Indicate `rcl_params_t` is or invalid.
+class InvalidParametersException : public std::exception
```

Comment:
exception type should be moved to `rclcpp/exceptions.hpp`

Reply:
Moved plus inherit from runtime_error in b415df7ed6f52c35569d973e9afaad32f05773c7